src/transport/SecureSessionMgr.h

```diff
@@ -114,15 +114,24 @@ class DLL_EXPORT SecureSessionMgrDelegate
 class DLL_EXPORT SecureSessionMgr : public Mdns::ResolveDelegate, public TransportMgrDelegate
 {
 public:
+    SecureSessionMgr();
+    ~SecureSessionMgr() override;
+
     /**
      * @brief
-     *   Send a message to a currently connected peer
+     *   Send a message to a currently connected peer.
+     *
+     * @details
+     *   msgBuf->Start() points to the packet being transmitted and msgBuf->DataLength()
+     *   is its length in octets. In case parameter bufferRetainSlot is not nullptr. Those
+     *   values are stored before sending the packet and rewinded back after sending the
+     *   packet so that msgBuf can be re-used for re-transmission.
```

Comment:
```suggestion
     *   msgBuf contains the data to be transmitted.  If bufferRetainSlot is not null and this function
     *   returns success, the encrypted data that was sent, as well as various other information needed
     *   to retransmit it, will be stored in *bufferRetainSlot.
```
Or so.  The key parts are:

1) The fact that we are rewinding is an implementation detail.  The important part is that we are storing the data in `*bufferRetainSlot`; we could be allocating a new `PacketBuffer` to do that for all the caller cares.
2) The data that can be retransmitted is in `*bufferRetainSlot`, _not_ in `msgBuf`.  `msgBuf` is being consumed by this function.